viewmodels/src/jsMain/kotlin/com/mirego/trikot/viewmodels/lifecycle/ApplicationStatePublisher.kt

```diff
@@ -1,7 +1,28 @@
 package com.mirego.trikot.viewmodels.lifecycle
 
 import com.mirego.trikot.streams.reactive.BehaviorSubjectImpl
+import kotlinx.browser.document
 import org.reactivestreams.Publisher
 
 actual class ApplicationStatePublisher :
-    BehaviorSubjectImpl<ApplicationState>(ApplicationState.FOREGROUND), Publisher<ApplicationState>
+    BehaviorSubjectImpl<ApplicationState>(ApplicationState.FOREGROUND),
```

Comment:
I think we should use a ColdPublisher instead, where we would `addEventListener` only on the first subscription. We could also remove this listener when subscriptions count drops back to zero.

Reply:
> Also, I would not dispatch the initial ApplicationState.FOREGROUND value and wait for the eventListener to trigger that first value.

Not sure on this one, I will try it out, but event listeners on the web are not like publishers and never triggers if the change does not occurs, which would let us without initial state until a background event is received.

Reply:
Updated to use the `firstSubscription` and `noSubscription`, works wonder!

No sure on how to proceed to use a ColdPublisher? Whenever you have time, can you educate me in Slack? thanks!